### ISSUE TITLE
`FusionSegmenterCombineReductionsCycleRepro_CUDA` is failing

### ISSUE THREAD

**Issue by zasdfgbnm** [NVIDIA/Fuser#477]
```c++
[ RUN      ] NVFuserTest.FusionSegmenterCombineReductionsCycleRepro_CUDA
unknown file: Failure
C++ exception with description "Input to expr is invalid,  it was not found in the active container.
Exception raised from assertInContainer at /home/gaoxiang/Fuser2/csrc/ir/container.h:54 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x7f6470ba85b7 in /home/gaoxiang/.local/lib/python3.11/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::string const&) + 0x64 (0x7f6470b66719 in /home/gaoxiang/.local/lib/python3.11/site-packages/torch/lib/libc10.so)
frame #2: nvfuser::Fusion::registerExpr(nvfuser::Expr*) + 0x395 (0x7f6470f0e1f5 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #3: nvfuser::IrBuilder::newArithmeticExpr(nvfuser::BinaryOpType, nvfuser::Val*, nvfuser::Val*) + 0x1e8 (0x7f6470f55ff8 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #4: nvfuser::vectorize_helper::ContiguousInnerDimensionsMapper::getContigMergeOfInnerSize(nvfuser::TensorView*) + 0x31b (0x7f647119ebbb in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #5: nvfuser::vectorize_helper::ContiguousInnerDimensionsMapper::getTvToContigMergeOfInnerSizeMap() + 0x7e (0x7f647119f1be in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #6: <unknown function> + 0x5a2e89 (0x7f64711a2e89 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #7: <unknown function> + 0x5a320b (0x7f64711a320b in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #8: nvfuser::HeuristicSummaryEntry<nvfuser::HeuristicCompileTime::TvToContigInnerSizeMaps>::HeuristicSummaryEntry(nvfuser::HeuristicSummary*, std::function<std::unique_ptr<std::vector<std::unordered_map<nvfuser::TensorView*, nvfuser::Val*, std::hash<nvfuser::TensorView*>, std::equal_to<nvfuser::TensorView*>, std::allocator<std::pair<nvfuser::TensorView* const, nvfuser::Val*> > >, std::allocator<std::unordered_map<nvfuser::TensorView*, nvfuser::Val*, std::hash<nvfuser::TensorView*>, std::equal_to<nvfuser::TensorView*>, std::allocator<std::pair<nvfuser::TensorView* const, nvfuser::Val*> > > > >, std::default_delete<std::vector<std::unordered_map<nvfuser::TensorView*, nvfuser::Val*, std::hash<nvfuser::TensorView*>, std::equal_to<nvfuser::TensorView*>, std::allocator<std::pair<nvfuser::TensorView* const, nvfuser::Val*> > >, std::allocator<std::unordered_map<nvfuser::TensorView*, nvfuser::Val*, std::hash<nvfuser::TensorView*>, std::equal_to<nvfuser::TensorView*>, std::allocator<std::pair<nvfuser::TensorView* const, nvfuser::Val*> > > > > > > ()>) + 0x60 (0x7f6471186e50 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #9: nvfuser::vectorize_helper::getVectorizationFactor(nvfuser::SchedulerRuntimeInfo&, nvfuser::TensorView*, nvfuser::HeuristicSummary*, int) + 0xe8 (0x7f647119e278 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #10: <unknown function> + 0x577771 (0x7f6471177771 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #11: nvfuser::SchedulerEntry::proposeHeuristics(nvfuser::Fusion*, nvfuser::SchedulerRuntimeInfo&) + 0x5c (0x7f64711812dc in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #12: <unknown function> + 0x5ff44c (0x7f64711ff44c in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #13: nvfuser::SegmentCandidateFinder::codeGenSupportedMerge(nvfuser::SegmentedGroup*, nvfuser::SegmentedGroup*) + 0x136 (0x7f64711ff596 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #14: nvfuser::SegmentCandidateFinder::trySetUpMerge(nvfuser::SegmentedGroup*, std::vector<nvfuser::SegmentedGroup::NeighborGroup, std::allocator<nvfuser::SegmentedGroup::NeighborGroup> >) + 0x90 (0x7f64711ff760 in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #15: nvfuser::SegmentCandidateFinder::findSegments() + 0x4fa (0x7f647120acaa in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #16: nvfuser::SegmentCandidateFinder::SegmentCandidateFinder(std::unique_ptr<nvfuser::Fusion, std::default_delete<nvfuser::Fusion> >, nvfuser::KernelArgumentHolder const&, nvfuser::SegmentCandidateFinderOptions) + 0x41a (0x7f647120b5ea in /home/gaoxiang/Fuser2/build/libnvfuser_codegen.so)
frame #17: nvfuser::SegmentCandidateFinder::segment(nvfuser::Fusion const*, nvfuser::KernelArgumentHolder const&, nvfuser::SegmentCandidateFinderOptions) + 0xea (0x556b4432300a in ./build/nvfuser_tests)
frame #18: <unknown function> + 0x27fded (0x556b44298ded in ./build/nvfuser_tests)
frame #19: <unknown function> + 0x5a6587 (0x556b445bf587 in ./build/nvfuser_tests)
frame #20: <unknown function> + 0x59ad3d (0x556b445b3d3d in ./build/nvfuser_tests)
frame #21: <unknown function> + 0x59af0d (0x556b445b3f0d in ./build/nvfuser_tests)
frame #22: <unknown function> + 0x59b657 (0x556b445b4657 in ./build/nvfuser_tests)
frame #23: <unknown function> + 0x59c004 (0x556b445b5004 in ./build/nvfuser_tests)
frame #24: <unknown function> + 0x5a6af7 (0x556b445bfaf7 in ./build/nvfuser_tests)
frame #25: <unknown function> + 0x59b014 (0x556b445b4014 in ./build/nvfuser_tests)
frame #26: <unknown function> + 0x14418a (0x556b4415d18a in ./build/nvfuser_tests)
frame #27: <unknown function> + 0x23850 (0x7f640f839850 in /usr/lib/libc.so.6)
frame #28: __libc_start_main + 0x8a (0x7f640f83990a in /usr/lib/libc.so.6)
frame #29: <unknown function> + 0x178355 (0x556b44191355 in ./build/nvfuser_tests)
" thrown in the test body.
[  FAILED  ] NVFuserTest.FusionSegmenterCombineReductionsCycleRepro_CUDA (2909 ms)

```

**Comment by jjsjann123** [comment]
bisect it to this commit (#467) which introduced the failure.

```
commit 1f09cd3ace6f6456f06a9cd48ffdce1c8ae2abbd (HEAD)
Author: Liqiang Lu <116412316+liqiangxl@users.noreply.github.com>
Date:   Fri Jun 9 15:31:40 2023 -0400

    avoid 0 batch in layer norm backward (#467)

    Fix #465, the vectorization factor was not correctly calculated in `checkCombinedReductionShape`
```

The PR changes `/csrc/scheduler/registry.cpp`
```
-      if (!checkCombinedReductionShape(runtime_info, reduction_tvs)) {
+      // get vectorize_factor, same process to that in getPersistentHeuristics
+      auto reduced_tv = ir_utils::getSoleProducerTv(first_inner_reduction_tv);
+      auto properties = scheduler_utils::getReductionProperties(
+          fusion, runtime_info, first_inner_reduction_tv);
+      const auto vectorize_factor = vectorize_helper::getVectorizationFactor(
+          runtime_info,
+          reduced_tv,
+          data_cache,
+          (int)(reduced_tv->nDims() - properties.inner_most_dimension_ndims));
+      if (!checkCombinedReductionShape(
+              runtime_info, reduction_tvs, (int64_t)vectorize_factor)) {
```

Where `getVectorizationFactor` triggers the assertion. To be clear, we never really guard the existing fusion when checking scheduler, so any modification to the graph is invalid.

**Comment by zasdfgbnm** [comment]
I see. Thanks for looking into it!
